### PR TITLE
Update espresso-ci scripts

### LINKED ISSUE #3612: Update espresso-ci bot with new GET syntax
GitHub has deprecated the GET syntax we use for espresso-ci ([Deprecating API authentication through query parameters](https://developer.github.com/changes/2020-02-10-deprecating-auth-through-query-param/)). We currently get a monthly email reminder, then starting September 30th the GitHub API will randomly fail to incentivize the migration to the new GET syntax.

### PR DESCRIPTION
**Pull request by jngrad**
Description of changes:
- use the new GET syntax for the GitHub API (fixes #3612)
- print a clear error message and exit Python with a non-zero return value when the GitHub API returns an HTTP error code (partial fix for #2832)

### CODE REVIEW

File: src/core/statistics.hpp [hunk 1]
```diff
@@ -59,7 +59,7 @@ int get_n_part_conf();
 double mindist(PartCfg &, IntList const &set1, IntList const &set2);
 
 /** Find all particles within a given radius @p r_catch around a position.
- *  @param partCfg    @copybrief PartCfg
+ *  @param partCfg    @copybrieef PartCfg
```
Comment (fweik) [inline src/core/statistics.hpp]: typo

**Review by KaiSzuttor — APPROVED** [review]
LGTM modulo tmp changes

**Comment by jngrad** [comment]
I should have seen that coming: we need a chained if-else to test all possible names for clang-format and we need to use `clang-format-9` instead of 6. The difference is minimal (only 2 files) but it's a circular change with version 6, i.e. once formatted with version 9, formatting it with the version 6 installed at the ICP will revert the change... So we're stuck with the Ubuntu 16.04 image until the ICP migrates to Ubuntu 20.04, which might take us months.

**Comment by KaiSzuttor** [comment]
I already approved

**Comment by fweik** [comment]
Yes, but there were new changes after, did you look at them?

**Comment by fweik** [comment]
Also, since when is the approval not removed if new changes are pushed?

**Comment by jngrad** [comment]
> Also, since when is the approval not removed if new changes are pushed?

I realized that behavior several weeks ago and found it strange too, but did not question it. The "marked this pull request as ready for review 11 hours ago" message it posted automatically when removing the draft status of a PR. I did not add the `automerge` label immediately because we had to wait for the PR to complete to make sure we could trigger the webhook for a successful CI pipeline.

**Comment by fweik** [comment]
It's a setting in the branch protection rules. Not sure if this conflicts with kodiak's merge ~~request~~ commits, if so it has to be deactivated again.

**Pull request merged**